app/src/main/java/de/tum/in/tumcampusapp/component/ui/tufilm/KinoDetailsFragment.java

```diff
@@ -116,32 +117,49 @@ private void showEventTicketDetails(Event event) {
 
     private void initBuyOrShowTicket(Event event) {
         MaterialButton ticketButton = rootView.findViewById(R.id.buyTicketButton);
-        if (ticketsLocalRepo.getTicketCount(event) != 0) {
-            ticketButton.setText(R.string.show_ticket);
-            ticketButton.setVisibility(View.VISIBLE);
+        int ticketBoughtCount = ticketsLocalRepo.getTicketCount(event);
+
+        if (ticketBoughtCount > 0) {
+            ticketButton.setText(getResources().getQuantityString(R.plurals.show_tickets, ticketBoughtCount));
             ticketButton.setOnClickListener(view -> {
                 Intent intent = new Intent(getContext(), ShowTicketActivity.class);
                 intent.putExtra(KEY_EVENT_ID, event.getId());
                 startActivity(intent);
             });
         } else if (!EventHelper.Companion.isEventImminent(event)) {
             ticketButton.setText(R.string.buy_ticket);
-            ticketButton.setVisibility(View.VISIBLE);
             ticketButton.setOnClickListener(
                     view -> EventHelper.Companion.buyTicket(this.event, ticketButton, getContext()));
         }
     }
 
-    private void showTicketCount(@Nullable Integer count) {
-        String text;
+    private void showTicketCount(@Nullable TicketStatus status) {
+        TextView remainingTicketsTextView = rootView.findViewById(R.id.remainingTicketsTextView);
+        TextView remainingTicketsHeaderView = rootView.findViewById(R.id.remainingTicketsHeaderTextView);
+        MaterialButton ticketButton = rootView.findViewById(R.id.buyTicketButton);
+        int ticketBoughtCount = ticketsLocalRepo.getTicketCount(event);
 
-        if (count != null) {
-            text = String.format(Locale.getDefault(), "%d", count);
+        if(EventHelper.Companion.isEventImminent(event)) {
+            ticketButton.setVisibility((ticketBoughtCount <= 0) ? View.GONE : View.VISIBLE);
+            remainingTicketsHeaderView.setVisibility(View.GONE);
+            remainingTicketsTextView.setVisibility(View.GONE);
         } else {
-            text = getString(R.string.unknown);
+            if (status == null || status.getContingent() < 0) {
```

Comment:
This condition could be a helper function of TicketStatus. Probably no way to get rid of the null check I guess.